packages/sync/src/PersistentStore.ts

```diff
@@ -9,3 +8,199 @@ export interface PersistentStore<T> {
 }
 
 export type PersistedData = string | null | object;
+
+export class StorageWrapper implements PersistentStore<PersistedData> {
+  public storage: PersistentStore<PersistedData>;
+  private crypto = window.crypto;
+  private masterKey: CryptoKey | undefined;
+  private password: string;
+
+  constructor(storage: PersistentStore<PersistedData>, encryptionPassword: string) {
+    this.password = encryptionPassword;
+    this.storage = storage;
+    this.init();
```

Comment:
not sure if it's a good idea to call `init` here and but not wait for it to complete. It could cause some issues depending on how it is invoked. I suggest move the `init` call into `getItem` and `setItem` method.